app/src/main/java/com/example/dinhh/soundscape/presentation/screens/sounds/SoundActivity.kt

```diff
@@ -131,6 +138,10 @@ class SoundActivity : AppCompatActivity(), SoundAdapterViewHolderClicks {
 
     private fun goToMixer() {
         finish()
+        val intent = Intent(this, MixerActivity::class.java)
+        if(!cameFromPopup) {
+            startActivity(intent)
+        }
```

Comment:
Without that function, it adds a second mixer screen when selecting the sound from the popup window and you need to press back button twice to get out from there.

Reply:
I think finish() is enough. Try to delete these code to see if it works. If not then I agree with your answer

```
val intent = Intent(this, MixerActivity::class.java)
        if(!cameFromPopup) {
            startActivity(intent)
        }
```

Reply:
In that case it would not go to the mixer fragment when you click the "add sound" button from the library without going to the Mixer first. Then it just deletes the sounds list screen and goes back to the home screen.

Reply:
Accepted